File: samples/cs/08objects/objects.cs
using System;

namespace Objects {
    interface Shape {
        float area();
        float perimeter(); 
    }

    class Rectangle : Shape {
        float height;
        float width;

        public Rectangle(float height, float width) {
            this.height = height;
            this.width = width;
        }

        public float area() {
            return this.height * this.width;
        }

        public float perimeter() {
            return this.height * 2 + this.width * 2;
        }

        public float getHeight() {
            return this.height;
        }

        public float getWidth() {
            return this.width;
        }
    }

    class Square : Rectangle {
        public Square(float side) : base(side, side) {}

        public float getSide() {
            return this.getHeight();
        }
    }

    class Triangle : Rectangle {
        float hypotenuse;

        public Triangle(float height, float width) : base(height, width) {
            this.hypotenuse = (float)Math.Sqrt(height * height + width * width);
        }

        public new float area() {
            return base.area() / 2;
        }

        public new float perimeter() {
            return base.perimeter() / 2 + this.hypotenuse;
        }

        public float getHypotenuse() {
            return this.hypotenuse;
        }
    }

    class Circle : Shape {
        float radius;

        public Circle(float radius) {
            this.radius = radius;
        }

        public float area() {
            return (float)Math.PI * this.radius * this.radius;
        }

        public float perimeter() {
            return 2 * (float)Math.PI * this.radius;
        }

        public float getRadius()  {
            return this.radius;
        }
    }

    class Trapezoid : Rectangle {
        Triangle triangle;

        public Trapezoid(float height, float totalWidth, float triangleWidth): base(height, totalWidth) {
            this.triangle = new Triangle(height, triangleWidth);
        }

        public new float area() {
            return base.area() - this.triangle.area();
        }

        public new float perimeter() {
            return base.perimeter() - base.getHeight() - this.triangle.getWidth() + this.triangle.getHypotenuse();
        }
    }

    class Exec {
        static void Main(string[] args) {
            Rectangle r = new Rectangle(5, 2);
            Console.WriteLine("[Rectangle] A: {0}, P: {1}", r.area(), r.perimeter());
            Square s = new Square(4);
            Console.WriteLine("[Square]    A: {0}, P: {1}", s.area(), s.perimeter());
            Triangle t = new Triangle(3, 2);
            Console.WriteLine("[Triangle]  A: {0}, P: {1}", t.area(), t.perimeter());
            Circle c = new Circle(4);
            Console.WriteLine("[Circle]    A: {0}, P: {1}", c.area(), c.perimeter());
            Trapezoid p = new Trapezoid(2, 5, 2);
            Console.WriteLine("[Trapezoid] A: {0}, P: {1}", p.area(), p.perimeter());
        }
    }
}

File: samples/dart/08objects/objects.dart
import "dart:math";

abstract class Shape {
    double area();
    double perimeter();
}

class Rectangle extends Shape {
    double height;
    double width;

    Rectangle(this.height, this.width);

    double area() {
        return this.width * this.height;
    }

    double perimeter() {
        return this.width * 2 + this.height * 2;
    }

    double getWidth() {
        return this.width;
    }

    double getHeight() {
        return this.height;
    }
}

class Square extends Rectangle {
    Square(double side) : super(side, side);

    double getSide() {
        return this.height;
    }
}

class Triangle extends Rectangle {
    double hypotenuse;

    Triangle(double height, double width) : super(height, width) {
        this.hypotenuse = sqrt(height * height + width * width);
    }

    double area() {
        return super.area() / 2;
    }

    double perimeter() {
        return super.perimeter() / 2 + this.hypotenuse;
    }
}

class Circle extends Shape {
  double radius;

  Circle(this.radius);

  double area() {
    return pi * this.radius * this.radius;
  }

  double perimeter() {
    return 2 * pi * this.radius;
  }
}

void main() {
    Shape r = Rectangle(5, 2);
    print("[Rectange] A: ${r.area()} P: ${r.perimeter()}");
    Shape s = Square(4);
    print("[Square]   A: ${s.area()} P: ${s.perimeter()}");
    Triangle t = Triangle(4, 2);
    print("[Triangle] A: ${t.area()} P: ${t.perimeter()}");
    Circle c = Circle(3);
    print("[Circle]   A: ${c.area()} P: ${c.perimeter()}");
}